### PR TITLE
Devcontainer Tweaks for Running Locally

### PR DESCRIPTION
**Pull request by ikegentz**
## Description
Was opening the docs for the first time the other day (had never worked in this repo before so have a "vanilla, clean" install). Was not able to access the docs in my browser when using the devcontainer. This is a two-fold issue:
- No port is specified to be fowarded in the devcontainer configuration
- Docusaurus needed to be configured to listen on other hosts besides `localhost`. In this case, `0.0.0.0` (any host) makes sense since the development server will only ever used locally, may as well listen on any address the devcontainer may be given

## Related Issue(s)
No related issue, this is just something I noticed when working on something else and figured it may as well get fixed

Fixes #

## How to test
Open the devcontainer on the `main` branch, you can't get to `localhost:3000/docs`. Now checkout this branch, and reopen the devcontainer, it will now be accessible

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
     },
     "customizations": {
-      "vscode": {
-        "extensions": [
-          "unifiedjs.vscode-mdx",
-          "ChrisChinchilla.vale-vscode"
-        ],
-        "settings": {}
-      }
+        "vscode": {
```
Comment (ikegentz) [inline .devcontainer/devcontainer.json]: I guess this file hasn't been edited in a while, bunch of auto-format spacing stuff

Comment (mrsimonemms) [reply .devcontainer/devcontainer.json]: I think you've got a 4 space indentation whereas I used a 2 space originally - the EditorConfig plugin _should_ solve this for you (which is installed to the devcontainer)

File: package.json [hunk 2]
```diff
@@ -49,4 +49,4 @@
   "engines": {
     "node": ">=20.0"
   }
-}
+}
```
Comment (mrsimonemms) [inline package.json]:
```suggestion
}

```

File: .devcontainer/devcontainer.json [hunk 3]
```diff
@@ -1,16 +1,19 @@
...
+            "extensions": [
+                "unifiedjs.vscode-mdx",
+                "ChrisChinchilla.vale-vscode"
+            ],
+            "settings": {}
+        }
     }
-}
+}
```
Comment (mrsimonemms) [inline .devcontainer/devcontainer.json]:
```suggestion
}

```

Please add a trailing line - should be solved by EditorConfig plugin

**Comment by fharper** [comment]
@ikegentz thanks for this PR. @mrsimonemms is the only one using Devcontainer so I'll let him review this (he already did). Note that he is on vacation starting today, so the final approval may not be until January.

**Review by mrsimonemms — COMMENTED** [review]
Just a couple of things to address

**Comment by ikegentz** [comment]
> Just a couple of things to address

Thanks for the quick review. You're correct, the `EditorConfig` extension needed configured to get the newlines saving and indentation set correctly.

I included the `.editorconfig` file that I needed to use to get this working, in the last commit. It seems like that should be included in the repo for consistency when using the devcontainer, but let me know what you think.

**Comment by ikegentz** [comment]
> @ikegentz thanks for this PR. @mrsimonemms is the only one using Devcontainer so I'll let him review this (he already did). Note that he is on vacation starting today, so the final approval may not be until January.

No worries thanks for the quick review. I think this is all set to go whenever

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (ikegentz) [reply .devcontainer/devcontainer.json]: Fixed this with the included `.editorconfig` file (last commit)

File: package.json (hunk 2 above)
Comment (ikegentz) [reply package.json]: Fixed this with the included .editorconfig file (last commit)

File: .devcontainer/devcontainer.json (hunk 3 above)
Comment (ikegentz) [reply .devcontainer/devcontainer.json]: Fixed this with the included .editorconfig file (last commit)

**Review by fharper — APPROVED** [review]
@mrsimonemms I'm approving since you are on vacation. Feel free to update anything once you're back.

Thanks @ikegentz 🎉

**Pull request merged**